Entry list

donacrisante opened the pull request.

donacrisante commented:
Quick info: I also added a Favourite button/icon, a Favourite Tab and a favourite toggle function in this US. My goal was also to get all entries displayed I you click on the "Recent journeys" tab and only the favourites when I click on the "Favourites" tab. The amount of entries and of the favourites should also be displayed.

ninaassmann commented on components/Divider.js:
```diff
@@ -0,0 +1,13 @@
+  margin: 0;
+  height: 1px;
+  background: color-water-10;
+`;
```
you make a styled hr here but you are using a normal one
i guess you need to change the hr in the return to Hr

donacrisante replied:
Hi Nina, thanks for the feedback! Good point! I had totally overlooked it. Thanks

ninaassmann commented on components/EntryList.js:
```diff
@@ -0,0 +1,55 @@
+        ))}
+      </div>
+      </section>
+      <button type="submit"> + </button>
```
this button does nothing, I guess the form you open
so maybe use a Link component with href="/"
when it should look like a button yu can use a styled component for that

donacrisante commented:
I had already sent my PR to the code review on Friday, but since I still discovered an error and have now fixed it, I am sending my PR to the code review again. The error was that my CO2 calculation worked, but the displayed result in the entry list was the same for all modes of transport. It was not updated depending on which means of transport I had specified. Now it should work.

Kheirah commented on components/EntryList.js:
```diff
@@ -0,0 +1,55 @@
+import { Fragment } from "react";
+import Tab from "./Tab";
+import Tabs from "./Tabs";
+/* import Badge from "./Badge"; */
```
Did you accidentally comment this out?
You are using an element called `badge` (in lowercase) in line 26 and 33. You probably want to use the actual `Badge` component for which you would need the import.

donacrisante replied:
Yes, I wanted to test something with this Badge and forgot to uncomment it. I did it now.

Kheirah commented on components/EntryForm.js:
```diff
@@ -50,23 +22,59 @@ export default function EntryForm() {
+    const fuel = newEntry.fuel;
+    const result = handleCalculateCo2(transport, km, fuel);
+    newEntry.result = result;
+    console.log(newEntry);
+    console.log(result);
```
Remove `console.logs` when you're done inspecting the behavior of your code.

donacrisante replied:
I have removed them.

Kheirah commented on pages/api/hello.js:
You can delete this file. It's included in any new project, but you don't need it.

donacrisante replied:
Done.

Thanks for your feedback.

Kheirah commented on pages/_app.js:
```diff
@@ -1,14 +1,63 @@
+    defaultValue: "all",
+    });
+
+  const [isFavourite, setIsFavourite] = useLocalStorageState("favourite", {
+    defaultValue: "false",
+  });
```
Mhh, are you sure you need this state? I can't see where you are using it. And you don't need it to be honest. I think you can remove it and remove `isFavorite` from the being passed down via props as well.

Kheirah reviewed (approved):
Very good!

Have a look at the state you're not really using and see if you can remove it since it's probably not necessary to keep around.

donacrisante commented:
I had sent my code to QA this morning, but the reviewer noticed that when I enter a number of kilometers with a comma, the result of the CO2 calculation shows NaN. So, to fix this error I added a regular expression to the Km input field (in EntryForm component) that accepts single numbers and decimal numbers with comma or point. I have also updated the handleCalculateCo2 function accordingly. Since I was already at validation attributes, I also added the attribute "required" and a regular expression for the two other input fields (start, destination).

Kheirah reviewed (approved):
Nice!

Patterns are a good way to check the entered value in inputs, **but only when the submit event is fired.**

Currently, you have a "calculate" button which _does not check_ the entered value. It's okay for now because I know you're planning to remove that button anyway. And then you're left with the submit button only which will check the values when clicked.
